source/macros.tex

```diff
@@ -316,6 +316,10 @@
 %% Produces 9 output characters.
 \newcommand{\commentellip}{\tcode{/* ...\ */}}
 
+%% Concepts
+\newcommand{\oldconcept}[1]{\textit{Cpp98#1}}
```

Comment:
This seems wrong to me. For a start, many of the things we'll be giving these names to did not exist in C++98 (why would we call things *Cpp98MoveConstrutible* or *Cpp98CopyInsertible* when C++98 had no idea that such things existed?). Also, C++98 doesn't exist any more, it was replaced by C++03. Using Cpp17 as the prefix instead would help with that...

But I'm also generally very unhappy about having two different definitions for the same term that are distinguished, effectively, by one of then being labeled "Old". Do we have a plan for getting rid of the old names before we ship the C++20 IS? If not, I think we should consider at least adding some introductory verbiage somewhere cautioning the reader that we have (eg) two things called CopyConstructible and they are not necessarily the same. Actually, that's wrong, we have three -- *Cpp98CopyConstructible*, `CopyConstructible<T>`, and `is_copy_constructible_v<T>` are all different. :-(

Reply:
I'm happy to change the prefix to `Cpp17`, or nearly anything else - no one in LWG or LEWG review has yet expressed concern about or preference for a different prefix.

Getting rid of the old names would necessitate jettisoning the majority of the Standard Library which is not likely to happen for C++20. We could make the names more different, but that would actually be counterproductive: the new names for the "old" terms must be similar to avoid confusion, and the names for the new concepts should be similar to the name of the old term it is intended to supplant.

Reply:
Perhaps just some words in the library front matter about the formatting convention would help? (Analogous to [syntax] in the language front matter.)

I've also realized that we have four different things called copy constructible, since the `CopyConstructible` concept has both syntactic requirements and semantic ones. Speaking of which: have we concluded how we should be specifying the syntactic versus semantic requirements yet? I like using "models" for the latter, and I'd suggest we use "satisfies" for the former (matching the core language definition) or the `Concept<X>` is `true` formulation suggested on the lib reflector.

Reply:
> I've also realized that we have four different things called copy constructible, since the `CopyConstructible` concept has both syntactic requirements and semantic ones.

As does `Cpp17CopyConstructible` - make it five. And yes, that usage of "models" and "satisfies" is exactly what I'm working toward in the merged Ranges proposal - it is *far* clearer than the current "satisfies the syntactic/semantic requirements" mismash. I'll submit an editorial cleanup for uses in the working draft once I'm sure LWG is happy with the formulation.

Reply:
I've pushed a commit to change the slug to *Cpp17*, and another adding a description of the typographic convention used for requirement names to [structure.requirements].